client/app/bundles/course/assessment/submission/actions/index.js

```diff
@@ -241,6 +242,72 @@ export function resetAnswer(submissionId, answerId, questionId) {
   };
 }
 
+export function deleteFile(submissionId, answerId, fileId, answerFields) {
+  const answers = Object.values(answerFields);
+  const payload = { submission: { answers } };
+
+  return (dispatch) => {
+    dispatch({ type: actionTypes.DELETE_FILE_REQUEST });
+
+    return CourseAPI.assessment.submissions.update(submissionId, payload)
+      .then(response => response.data)
+      .then((data) => {
+        if (data.redirect_url && data.format === 'html') {
+          window.location = data.redirect_url;
+        }
+        dispatch({ type: actionTypes.DELETE_FILE_SUCCESS, payload: data });
+        dispatch(reset(formNames.SUBMISSION));
+        dispatch(setNotification(translations.deleteFileSuccess));
+      })
+      .catch((error) => {
+        dispatch({ type: actionTypes.DELETE_FILE_FAILURE });
+        dispatch(setNotification(translations.deleteFileFailure, buildErrorMessage(error)));
+      });
+  };
+}
+
+// Used to ensure that only java files can be uploaded.
+function validateFiles(files) {
+  const regex = new RegExp('\\.java');
+  return files.filter(file => !regex.test(file.filename)).length === 0;
+}
+
+export function importFiles(submissionId, answerId, answerFields) {
+  const answers = Object.values(answerFields);
+  const files = answerFields[answerId].files_attributes;
+  const payload = { submission: { answers } };
+  /* eslint-disable consistent-return */
+  return (dispatch) => {
+    dispatch({ type: actionTypes.IMPORT_FILES_REQUEST });
+
+    if (!validateFiles(files)) {
+      dispatch({ type: actionTypes.IMPORT_FILES_FAILURE });
+      dispatch(setNotification(translations.invalidFileUpload));
+    } else {
+      return CourseAPI.assessment.submissions.update(submissionId, payload)
```

Comment:
I think this `return` can be removed. Then we can remove the ESLint disable statements. But if we remove `return` here we should do so for all other actions also for consistency. Maybe this should be done in a future PR instead. @weiqingtoh what do you think?

Reply:
Referencing off this (origin of the rule) -> https://github.com/eslint/eslint/issues/481

> A rule like this exists in JavaScriptLint. It's helpful for people coming from strongly typed languages. It also nice to know that the function will either always return a value, or never do so.

I think as long as the reader can easily identify the whether the function will always return a value or not, that is fine. So we can either: 
 1. Disable the lint and write some documentation; or
 1. Use the lint. 

Of course the best preference would be to use the lint and have documentation lah 😛